tests/acceptance/restrict-content/RestrictContentProductCPTCest.php

```diff
@@ -0,0 +1,341 @@
+	{
+		// Activate ConvertKit plugin.
+		$I->activateConvertKitPlugin($I);
+
+		// Create a public Custom Post Type called Articles, using the Custom Post Type UI Plugin.
+		$I->registerCustomPostType($I, 'article', 'Articles', 'Article');
+
+		// Create a non-public Custom Post Type called Private, using the Custom Post Type UI Plugin.
+		$I->registerCustomPostType($I, 'private', 'Private', 'Private', false);
+	}
+
+	/**
+	 * Test that content is not restricted when not configured on a WordPress CPT.
+	 *
+	 * @since   2.4.3
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function testRestrictContentWhenDisabled(AcceptanceTester $I)
+	{
+		// Setup ConvertKit Plugin, disabling JS.
+		$I->setupConvertKitPluginDisableJS($I);
+
+		// Add the CPT using the Gutenberg editor.
+		$I->addGutenbergPage($I, 'article', 'ConvertKit: Article: Restrict Content: Product');
+
+		// Add blocks.
+		$I->addGutenbergParagraphBlock($I, 'Visible content.');
+		$I->addGutenbergBlock($I, 'More', 'more');
+		$I->addGutenbergParagraphBlock($I, 'Member only content.');
+
+		// Publish Article.
+		$url = $I->publishGutenbergPage($I);
+
+		// Confirm that all content is displayed.
+		$I->amOnUrl($url);
+		$I->see('Visible content.');
+		$I->see('Member only content.');
+	}
+
+	/**
+	 * Test that no Restrict Content options are displayed when the Post Type
+	 * is not public.
+	 *
+	 * @since   2.4.3
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function testNoRestrictContentOnPrivateCPT(AcceptanceTester $I)
+	{
+		// Setup ConvertKit Plugin, disabling JS.
+		$I->setupConvertKitPluginDisableJS($I);
+
+		// Add the CPT using the Gutenberg editor.
+		$I->addGutenbergPage($I, 'private', 'ConvertKit: Private: Restrict Content');
+
+		// Check that the metabox is not displayed.
+		$I->dontSeeElementInDOM('#wp-convertkit-meta-box');
+
+		// Publish and view the Page on the frontend site.
+		$I->publishAndViewGutenbergPage($I);
+	}
+
+	/**
+	 * Test that restricting content by a Product specified in the CPT Settings works when
+	 * creating and viewing a new WordPress CPT.
+	 *
+	 * @since   2.4.3
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function testRestrictContentByProduct(AcceptanceTester $I)
+	{
+		// Setup ConvertKit Plugin, disabling JS.
+		$I->setupConvertKitPluginDisableJS($I);
+
+		// Add the CPT using the Gutenberg editor.
+		$I->addGutenbergPage($I, 'article', 'ConvertKit: Article: Restrict Content: Product');
+
+		// Configure metabox's Restrict Content setting = Product name.
+		$I->configureMetaboxSettings(
+			$I,
+			'wp-convertkit-meta-box',
+			[
+				'form'             => [ 'select2', 'None' ],
+				'restrict_content' => [ 'select2', $_ENV['CONVERTKIT_API_PRODUCT_NAME'] ],
+			]
+		);
+
+		// Add blocks.
+		$I->addGutenbergParagraphBlock($I, 'Visible content.');
+		$I->addGutenbergBlock($I, 'More', 'more');
+		$I->addGutenbergParagraphBlock($I, 'Member only content.');
```

Comment:
If this text is coming from us, I believe "member-only" is grammatically correct in this instance.

Reply:
This is within the tests, although there is one reference to 'Member only content' in a setting field description. Happy to change these in a separate PR if you think it's worthwhile.